src/drivers/dw/dma.c

```diff
@@ -1170,6 +1170,45 @@ static void dw_dma_process_block(struct dma_chan_data *chan,
 }
 #endif
 
+static int dw_dma_copy(struct dma *dma, int channel, int bytes, uint32_t flags)
+{
+	struct dma_pdata *p = dma_get_drvdata(dma);
+	struct dma_chan_data *chan = &p->chan[channel];
+	struct dw_lli2 *ll_uncached = cache_to_uncache(chan->lli_current);
+	struct dma_sg_elem next;
+
+	next.src = DMA_RELOAD_LLI;
+	next.dest = DMA_RELOAD_LLI;
+	next.size = bytes;
+
+	tracev_dwdma("dw-dma: %d channel %d copy", dma->plat_data.id,
+		     channel);
+
+	if (chan->status != COMP_STATE_ACTIVE) {
+		trace_dwdma_error("dw-dma: %d channel %d not running",
+				  dma->plat_data.id, channel);
+
+		/* skip if channel is not running */
+		return 0;
+	}
+
+	if (chan->cb)
+		chan->cb(chan->cb_data, DMA_IRQ_TYPE_BLOCK, &next);
```

Comment:
Could you clarify this a bit - is this some special case? Isn't the callback usually called upon DMA completion?

Reply:
It will pass copied size to the dai.